### ISSUE TITLE
Can't toggle individual element on and off after making a selection

### ISSUE THREAD

**Issue by elimist3** [Simonwep/selection#129]
#### What is the problem?
Clicking an element does not deselect it when another selection has already been made.
Maybe related to #91 

#### What is the current behavior?

Initially, toggling on and off by clicking works. After selecting a group of elements, toggling another element on and off no longer works.

[example](https://streamable.com/6anina)

#### Please provide the steps to reproduce and create a [CodeSandbox](https://codesandbox.io/).
Steps to reproduce:
1. Delete the clearing behaviour in onStart
2. Select a group of elements
3. Repeatedly tap on any other element that has not already been selected.

https://codesandbox.io/s/viselectreact-forked-wl0yb?file=/src/App.tsx

#### What is the expected behavior?
The expected behavior is that repeatedly tapping on any element regardless of whether it was selected or not should simply toggle it on and off.

#### Your environment:
Toolset: webpack
Version: v3.0.0-beta.7
Browser: Chrome/FF latest
OS: MacOS

**Comment by Simonwep** [comment]
Okay that's funny - what you want here is for some reason the exact opposite of #108. I'll mark this as improvement as it's more or less a change in behavior than a bug-fix :)

**Comment by elimist3** [comment]
Hmm, I think what #108 is describing is actually the correct behaviour for the following sequence of actions:

1. hold down CTRL / CMD
2. drag
3. release CTRL / CMD
4. click a single element

When you do it in that sequence, any active element should definitely stay active.

However in my case, the sequence of actions is more akin to:
1. hold down CTRL / CMD
2. drag
3. click any element (while still holding down CTRL / CMD)

In that case, clicking on an element (whether already selected or not) should toggle it. I believe that's the default behaviour of most OS's, for example on a [MacOS desktop](https://streamable.com/49iilt)

**Comment by Simonwep** [comment]
Okay - but just for clarification, looking at your codesandbox example you _don't_ want the same behavior as  as on MacOS? Because you always _keep_ the elements and don't de-select them in the "start" event:

```ts
// You normally want some code in here...
const onStart = ({ event, selection }: SelectionEvent) => {};
```
But I see the problem here - it should work without having the "start"-event part.

I'm gonna unmark this one since I'm not sure what needs to be changed to get both working (if that's even possible since I don't know the code you have in the `start` event).

**Comment by elimist3** [comment]
Sorry, I should have been more clear: I do indeed want the same behavior as on MacOS. 

> But I see the problem here - it should work without having the "start"-event part.

Exactly! I want to basically do a multi-select by default without having to hold down a modifier key. To clear previously selected elements, the user will either click on them individually or drag across a previously selected group of elements.

There is another library called [Selecto](https://github.com/daybrush/selecto) that has the [desired behaviour](https://daybrush.com/selecto/storybook/?path=/story/selecto--continue-to-select&knob-hitRate=0&knob-selectByClick=true&knob-selectFromInside=true&knob-continueSelect=true&knob-ratio=0), however its bundle is double the size of this library so I'd prefer to use this one :)

> I'm gonna unmark this one since I'm not sure what needs to be changed to get both working (if that's even possible since I don't know the code you have in the start event).

Right now I have no code in the `start` event. I could probably simulate the behaviour I'm looking for inside `onStart`, but I'm thinking that this is more of a core behaviour.

**Comment by Simonwep** [comment]
> Exactly! I want to basically do a multi-select by default without having to hold down a modifier key. To clear previously selected elements, the user will either click on them individually or drag across a previously selected group of elements.

Gotchya, yeah I know Selecto, I got the idea for providing framework-wrapper from it ^^

> Right now I have no code in the start event. I could probably simulate the behaviour I'm looking for inside onStart, but I'm thinking that this is more of a core behaviour.

Yeah... that's true. In this case I'd even consider this a bug or design-limitation which has to be fixed.... aaaand it's fixed. I'll release a new beta later today!

**Comment by Simonwep** [comment]
Now available in [v3.0.0-beta.8](https://github.com/Simonwep/selection/releases/tag/v3.0.0-beta.8), wanted to get this off my head ^^

**Comment by elimist3** [comment]
Awesome, it work perfectly now.

Cheers!
